code/controllers/subsystem/weather.dm

```diff
@@ -8,24 +8,66 @@ SUBSYSTEM_DEF(weather)
 	var/list/eligible_zlevels = list()
 	var/list/next_hit_by_zlevel = list() //Used by barometers to know when the next storm is coming
 
-/datum/controller/subsystem/weather/fire()
+/datum/controller/subsystem/weather/fire(resumed = FALSE)
 	// process active weather
-	for(var/V in processing)
-		var/datum/weather/our_event = V
-		if(our_event.aesthetic || our_event.stage != MAIN_STAGE)
+	for(var/datum/weather/weather_event in processing)
+		if(IS_WEATHER_AESTHETIC(weather_event.weather_flags) || weather_event.stage != MAIN_STAGE)
 			continue
```

Comment:
similar here why not check if the subsystem has any tasks it's asking for?

Reply:
They both are accomplishing the same thing. (since subsystems get determined via bitflags)

Applied via https://github.com/tgstation/tgstation/pull/89550/commits/af5f9c2bf5358b1730476e1504acf648f2fe639a